joern-cli/frontends/php2cpg/src/test/scala/io/joern/php2cpg/querying/TypeDeclTests.scala

```diff
@@ -369,13 +369,13 @@ class TypeDeclTests extends PhpCode2CpgFixture {
         |""".stripMargin)
 
     "parse methods in classes correctly" in {
-      inside(cpg.typeDecl.name("anon-class-\\d+").l) {
+      inside(cpg.typeDecl.name("Test0.php:<global>@anon-class-\\d+").l) {
         case anonClass0 :: anonClass1 :: Nil =>
           val List(memberX) = anonClass0.member.l
           memberX.code shouldBe "$x"
 
           val List(anonConstructor0) = anonClass0.method.name("__construct").l
-          anonConstructor0.fullName shouldBe "anon-class-0->__construct"
+          anonConstructor0.fullName shouldBe "Test0.php:<global>@anon-class-0->__construct"
```

Comment:
Is there some sort of special meaning of `@` and `->` in php? Why not stick to the `.` separator we use for most of the other frontends?

Reply:
I just used the `@` as a separator between the namespace and the anon-class name so we have `namespace@anon-class`. I can change this to be `.`

The `->` is what is always used when creating the fullName for the `init` call (i.e. the call to `__construct`) so that I would say we keep as is (`AstForExpressionCreator:847`)

Reply:
So there is no apparent reason as it looks. Please make a followup PR where only `.` instead of `@` and `->`.
This would also require adjusting the policy full names and the sptest expectation files.

Reply:
PHP uses `->` for member/method access. Since the `.` operator was already taken to mean string concat when OOP support was added. Makes sense to me to use the same in our names.

Reply:
@maltek @ml86 - should I go ahead with a new PR and make the changes to the `@` and `->`?

Reply:
@AndreiDreyer Please wait with such a PR. There are other questions regarding the full names which need to get sorted out first.